AutoMad/wordGrabber.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct  3 11:21:19 2024
+
+@author: dragon
+"""
+
+import os.path
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+import numpy as np
+import pandas as pd
+import madDictionary
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ["https://www.googleapis.com/auth/spreadsheets.readonly","https://www.googleapis.com/auth/documents"]
+
+# the dictionary and story list sheet.
+MADLIB_SHEET_ID = "1GKVnudQK-5QdoxBpt2KceHg6ZooyMYwU7fWAVFAyouw"
+WORDS = "Words!A1:z1000"
+STORIES = "Stories!A1:z1000"
+
+def getCreds():
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    if os.path.exists("token.json"):
+        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                "credentials.json", SCOPES
+            )
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open("token.json", "w") as token:
+            token.write(creds.to_json())
+    return creds
+
+def getStoryByName(name):
+    result = getGoogleSheet(MADLIB_SHEET_ID,STORIES)    
+    values=result['values']
+    df = pd.DataFrame(values)
+    df.columns = df.iloc[0]
+    df = df[1:]
+    for index,row in df.iterrows():
+        if row['storyname'] == name:
+            return row['urlString']
+    return None
+
+def getStories():
+    result = getGoogleSheet(MADLIB_SHEET_ID,STORIES)
+    values=result['values']
+    df = pd.DataFrame(values)
+    df.columns = df.iloc[0]
+    df = df[1:]
+    return df.storyname
```

Comment:
Can this be

```suggestion
    values=result['values']['storyname']
```

It looks like you may be falling into the pattern some data scientists do of reaching for pandas for everything. Pandas is great, and if it is doing some normalizing for you that you need for this, that is fine. If not, it makes the code harder to understand than it needs to be, and will waste a lot of compute cycles (not that that matters much)